File: test_files/policy-based-enforcement-modes.java
/*
Test: Policy-Based Enforcement Modes (Java)
Includes inline policy pragmas and suppression comments to simulate allow/deny/monitor modes.
*/

// guardrails: allow=hardcoded-secret reason="testing-exemption"
public class policy_based_enforcement_modes {
    public static final String ADMIN_PASSWORD = "hunter2";

    // guardrails: mode=monitor
    public static String maybeInsecureAction() {
        // This action is allowed in monitor mode but should be blocked in enforce mode
        return "performed risky action";
    }

    // guardrails: deny=unapproved-library
    // Simulated import that should be blocked in enforce mode
    // import com.unapproved.internal;

    public static void main(String[] args) {
        System.out.println("policy-based-enforcement-modes Java test loaded");
    }
}

File: test_files/policy-based-enforcement-modes.py
"""
Test: Policy-Based Enforcement Modes
Contains annotations and comments to simulate allow/deny/monitor modes and rule suppression pragmas.
"""

# Example of inline policy pragma to suppress a rule (should be recognized by policy engine)
# guardrails: allow=hardcoded-secret reason="testing-exemption"
ADMIN_PASSWORD = "hunter2"

# Example of monitor-only marker
# guardrails: mode=monitor
def maybe_insecure_action():
    # This action is allowed in monitor mode but should be blocked in enforce mode
    return 'performed risky action'

# Example of deny marker
# guardrails: deny=unapproved-library
import unapproved_internal_lib  # this import should be blocked in enforce mode

if __name__ == '__main__':
    print('policy-based-enforcement-modes test file loaded')
